### DISCUSSION TITLE
How to move contents of public dir to app.baseUrl?

### DISCUSSION THREAD

**Discussion by ramonakira** [nuxt/nuxt#17227] [Questions]
Hi,

I'm trying to create build where all files are isolated within a subdirectory and we cannot put anything in the root. So I specify `app.baseURL` in nuxt.config.ts to point to my subdirectory and `buildAssetsDir` points to `/<subdirectory>/_nuxt/`. This all seems to work fine. Now I have some javascript files in my public-directory that - after a build - are still in the root directory. How do I make sure that the files from the public directory are moved to within the subdirectory?

Also, there is still a index.html generated on the root. How do I prevent this?

Thanks in advance!
Ramon

**Comment by mirumirumi** [comment]
Hi!

**If** by "directory" you are talking about URL paths in a website, then my response is as follows:

Create a subdirectory of any name in the `public/` directory.
That way, after the build (presumably you mean nuxt generate) you will still have `https://yourdomain.com/[created-directory]/yourfiles.*` .

===

**ELSE:**

Although our objectives are different, I, like you, would like to stop making the project root the main directory of the application.

I have actually succeeded in moving everything into the `src/` directory on a website I built: 
https://github.com/mirumirumi/mirumi-tech-frontend

Here is `nuxt.config.ts` :
https://github.com/mirumirumi/mirumi-tech-frontend/blob/main/nuxt.config.ts

Have a good day :)
